stream-parent/stream-client/src/main/java/org/interledger/stream/sender/StreamConnectionManager.java

```diff
@@ -0,0 +1,75 @@
+ *
+ * <p>For example, a Stream Sender that allows for simultaneous calls to `sendMoney` might guard its internal sequence
+ * to make it thread-safe, but another instance might accidentally be constructed at the same time. These two
+ * implementations' sequence might collide in this case.</p>
+ *
+ * <p>This implementation provides a JVM-wide location to open and close connections so that no two instances of a
+ * Stream sender or receiver accidentally use sequences that might overlap. While this implementation does not guard
+ * against two classes in _different_ JVMs using the same destination address and secret, and thus potentially
+ * colliding, this possibility is considered to be extremely remote since the chances of two different JVMs using the
+ * same shared secret remote. However, care should be taken when designing clustered STREAM senders and receivers
+ * because it is possible that certain implementations of Stream Receiver might return the same shared secret if queried
+ * more than once using the same receiver address.</p>
+ */
+public class StreamConnectionManager {
+
+  /**
+   * A {@link Map} of {@link StreamConnection} keyed by a {@link StreamConnectionId}. This ensures that while multiple
+   * {@link StreamSender#sendMoney} requests can operate in parallel, the underlying StreamConnection will guarantee a
+   * monotonically increasing sequence number so that parallel Streams acrros multiple instances of the same
+   * StreamConnection do not interfere with each other. per Connection (i.e., SharedSecret+destination address) may be
+   * executed at a single time.
+   */
+  private static final Map<StreamConnectionId, StreamConnection> connections = Maps.newConcurrentMap();
+
+  public StreamConnection openConnection(final StreamConnectionId streamConnectionId) {
+    Objects.requireNonNull(streamConnectionId);
+
+    // If the connection is already open, then return it. Otherwise, return a new Connection.
+    return connections.computeIfAbsent(streamConnectionId, StreamConnection::new);
+  }
+
+  /**
+   * <p>Close the {@link StreamConnection} identified by {@code streamConnectionId}. This method is thread-safe because
+   * the underlying {@link StreamConnection} is itself thread-safe, and will not allow emitted sequences that are
+   * invalid.</p>
+   *
+   * <p>One concurrency-related note is that there is a small moment of time after this method has been called, but
+   * where it has not yet processed the close request. During this time, an in-use Connection will process calls to
+   * {@link StreamConnection#nextSequence()} so long as the sequence is not too large. Ultimately it is the
+   * responsibility of {@link StreamConnection} to manage it effective Connection state.</p>
+   *
+   * @param streamConnectionId
+   *
+   * @return
+   */
+  public Optional<StreamConnection> closeConnection(final StreamConnectionId streamConnectionId) {
```

Comment:
Intellij says this method is never called. Seems like it should be.

Reply:
See my comment above. We'll revisit this as part of #291